From 45f409aa60c7c4a575b267134b3d51bdde75a377 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Wed, 22 Oct 2014 01:12:35 -0700
Subject: [PATCH] pyanaconda/pyudev.py: find udev lib while systemd enabled

By default, we search udev lib with name libudev.so.0,
but while systemd enabled, the udev lib name is
libudev.so.1

Try to search libudev.so.1 while libudev.so.0 not found.

Upstream-Status: Pending

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 pyanaconda/pyudev.py |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/pyanaconda/pyudev.py b/pyanaconda/pyudev.py
index acf5154..608c98b 100644
--- a/pyanaconda/pyudev.py
+++ b/pyanaconda/pyudev.py
@@ -32,6 +32,9 @@ def find_library(name, somajor=0):
 
 # find the udev library
 libudev = find_library(name="udev", somajor=0)
+if not libudev or not os.path.exists(libudev):
+    # find udev library while systemd enabled
+    libudev = find_library(name="udev", somajor=1)
 
 if not libudev or not os.path.exists(libudev):
     raise ImportError, "No library named %s" % libudev
-- 
1.7.1

